backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs

```diff
@@ -212,6 +239,15 @@ public async Task<ChallengeFullDto> EditChallengeAsync(ChallengeEditDto challeng
             return await GetChallengeFullDtoByIdAsync(challenge.Id);
         }
 
+        private async Task<BriefChallengeInfoDto> GetBriefChallengeInfoById(long challengeId)
+        {
+            var challenge = await _context.Challenges
+                .Include(x => x.Author)
+                .SingleOrDefaultAsync(challenge => challenge.Id == challengeId);
```

Comment:
Why we need SingleOrDefault here?
It requires more effort to execute than firstOrDefault()